Library/Homebrew/cmd/missing.rb

```diff
@@ -6,15 +6,40 @@ module Homebrew
   def missing
     return unless HOMEBREW_CELLAR.exist?
 
-    ff = if ARGV.named.empty?
+    formulae = if ARGV.named.empty?
       Formula.installed
     else
       ARGV.resolved_formulae
     end
+    if ARGV.json == "v1"
+      missing = print_missing_json(formulae)
+    else
+      missing = print_missing(formulae)
+    end
+    Homebrew.failed = missing.any?
+  end
+
+  def print_missing(formulae)
+    verbose = ($stdout.tty? || ARGV.verbose?) && !ARGV.flag?("--quiet")
+    s = Set.new
+    missing = Diagnostic.missing_deps(formulae) do |name, m|
+      m.each { |f| s << f }
+      if verbose
+        print "#{name}: " if formulae.size > 1
+        puts "#{m * " "}"
+      end
+    end
+    s.each { |f| puts f.full_name } unless verbose
```

Comment:
I'm in favor of `puts_columns` and sorted for sure. Order should be either simply alphabetical or core formulae followed by formulae from taps (see `brew list --full-name`; I'm afraid we're not very consistent with that across various commands).